### DISCUSSION TITLE
Replacing Valetudo on S5 Max

### DISCUSSION THREAD

**Discussion by lanzi69** [rand256/valetudo#438] [Ideas]
Hey @all,

I recently patched my S5 Max using a DustBuilder image based on a stripped Ubuntu with Valetudo 2021.04.0 (see [s5e-cheatsheet.txt](https://builder.dontvacuum.me/s5e-cheatsheet.txt) for details).

Because Valetudo doesn't support different languages I wanted to try using Valetudo RE.

I temporary renamed Valetudo, replaced it by Valetudo RE and after testing a bit I decided to replace Valetudo permanently.

Due to the above patching precedure the filesystem containing "/etc" is write protected so that the recommended installation procedure with starting Valetudo RE as an Upstart script does not work.

During the above mentioned patching process the "valetudo" binary is copied to "/mnt/data" and on device startup this binary is launched by the script "/mnt/reserve/_root.sh".

To get Valetudo RE running it is necessary to have the "valetudo" binary installed to the directory "/mnt/data/valetudo" where Valetudo RE also places its configuration.

So I connected to the vaccum via SSH and executed the following steps for replacement:

```
[root@rockrobo ~]# killall valetudo
[root@rockrobo ~]# rm /mnt/data/valetudo
[root@rockrobo ~]# mkdir /mnt/data/valetudo
[root@rockrobo ~]# cd /mnt/data/valetudo
[root@rockrobo valetudo]# wget https://github.com/rand256/valetudo/releases/download/0.10.7/valetudo.tar.gz
[root@rockrobo valetudo]# tar xfvz valetudo.tar.gz
[root@rockrobo valetudo]# rm  valetudo.tar.gz
[root@rockrobo valetudo]# sed -i -e 's/\/mnt\/data\/valetudo/\/mnt\/data\/valetudo\/valetudo/g' /mnt/reserve/_root.sh
[root@rockrobo valetudo]# sed -i -e 's/VALETUDO_CONFIG_PATH=\/mnt\/data\/valetudo_config\.json\ //g' /mnt/reserve/_root.sh
[root@rockrobo valetudo]# reboot
```

To ensure that the binary can be verified after extracting it would be a good idea to add an MD5 hash to the tarball. ;-)

Feel free to adapt my installation procedure to your Wiki.

Thanx & cu
Lanzi

**Comment by svn2208** [comment]
I basically did the same, but logging is not working. Maybe the binary output need to append to /mnt/data/valetudo/valetudo.log and linked to /var/log/upstart/valetudo.log within the _root.sh. What do you think?


here is an example; I am not an bash expert.
```
  rm /mnt/data/valetudo/valetudo.log
        /mnt/data/bin/valetudo >> /mnt/data/valetudo/valetudo.log 2>&1 &
        mkdir -p /var/log/upstart
        ln -s -f /mnt/data/valetudo/valetudo.log /var/log/upstart/valetudo.log

```

**Reply by lanzi69** [reply]
I didn't get logging to work so I didn't touch the redirect-part of the command line in "/mnt/reserve/_root.sh". :-P

Actually there is no file "/mnt/data/valetudo/valetudo.log" and the app doesn't create one, if launched without any redirect.

I scanned the "valetudo" binary for the string "/mnt/data/valetudo" and found the seemingly needed subdirectories "uploads" and "maps" but I don't know, what they are used for.

Wouldn't it be a better idea changing the logfile-path inside the app to "/mnt/data/valetudo/valetudo.log" instead of linking it to "/var/log/upstart/valetudo.log"?

Meanwhile it's a quite good way to get logging to work - thanx for the hint!

**Reply by svn2208** [reply]
You are right: Changing the path is the better way, but I guess, this path is because of a different firmware-mod and not the dustcloud tutorial. 

But logging is not so detailed as in the Hypfer version; also loglevels imho can't be alters.
